spark/src/main/scala/org/apache/spark/sql/delta/managedcommit/CommitStore.scala

```diff
@@ -106,6 +106,28 @@ trait CommitStore {
       logPath: Path,
       startVersion: Long,
       endVersion: Option[Long]): Unit
+
+  /**
+   * Determines whether this CommitStore is semantically equal to another CommitStore.
+   *
+   * Semantic equality is determined by each CommitStore implementation based on whether the two
+   * instances can be used interchangeably when invoking any of the CommitStore APIs, such as
+   * `commit`, `getCommits`, etc. For e.g., both the instances might be pointing to the same
+   * underlying endpoint.
+   */
+  def semanticEquals(other: CommitStore): Boolean
+}
+
+object CommitStore {
+  def semanticEquals(
```

Comment:
Can you clarify why we need this static method? Is it just a helper to simplify dealing with the options? Do we expect this to be used frequently?

Reply:
It's a helper method used in OptimisticTransaction. Inlining the logic there was making it harder to read the code.